lib/boa/class_methods.rb

```diff
@@ -35,7 +35,7 @@ def properties
     # @return [ClassMethods] the class method module
     #
     # @api public
-    sig { params(name: Symbol, class_type: Type::ClassType, options: Object).returns(T.self_type) }
+    sig { params(name: Symbol, class_type: Type::ClassType, options: T.untyped).returns(T.self_type) }
```

Comment:
__comment:__ Normally this "weakening" would be bad for type safety, but these options are being forwarded to other objects' constructors where they will be more tightly constrained anyway. The options, at this scope, need to be untyped to allow all the various combinations that are then checked by the Type constructors.